### DISCUSSION TITLE
Install Issue

### DISCUSSION THREAD

**Discussion by n8-410** [Security-Onion-Solutions/securityonion#8797] [General]
Upon completion of an install using ./so-setup iso  there is an issue with the logic in in so-setup line 1109.  It checks to see if there is any size to the root users mail folder.  If this size is > zero then the entire setup errors out and isn't cleaned up.  

Using an automated distributed managersearch install_type (based on automation/distributed-airgap-manager) I ended up with a few influxdb_size.log file doesn't exist in the root mail spool (which eventually got created created during the install process), but because of those early messages the setup process was not cleaned up and errored out at the very end. 

Process
1. modify automation/distributed-airgap-manager (set install_type=MANAGERSEARCH,INTERWEBS=AIRGAP)
2. cd /root/SecurityOnion/setup
3. ./so-setup iso distributed-airgap-manager
4. wait ....
5. Error at the end
6. cat /root/errors.log returns nothing
7. cat /var/spool/mail/root only shows old issues with  influxdb

Not sure if this could be removed entirely or if the user could be presented with the errors and asked if they want to proceed with clean up. or have a way to start the clean up/final setup steps after a user review of the errors. or a retry option (present the errors.log which is empty and the mail spool output if an iso image)

**Comment by n8-410** [comment]
I just noticed the default so-setup that launches when the user logs in on an unconfigured seconion install executes using the iso method as well.  So all users would run into this same error using the default setup mechanism

sudo /home/onion/SecurityOnion/setup/so-setup iso

**Comment by dougburks** [comment]
> Using an automated distributed managersearch install_type (based on automation/distributed-airgap-manager)

The automation files are for our own internal testing and we do not support them for any other use.

> I just noticed the default so-setup that launches when the user logs in on an unconfigured seconion install executes using the iso method as well. So all users would run into this same error using the default setup mechanism

I don't believe we've had any other reports of this, so this might be something specific to your configuration or environment.

**Reply by n8-410** [reply]
> The automatiion files are for our own internal testing and we do not support them for any other use.

That is unfortunate.  Would it make sense to add a README to that folder to say this is an unsupported internal testing feature? 
I have been trying to come up with an automated way of setting up networks for testing/training using a new security onion in a distributed setup with each network. Automation seemed to be a viable option as I can build the automation file as needed.  

An alternative would be to build the manager and then change the IP as needed (different networks have different IP space), but their doesn't seem to be an so_ file that will change the manager IP on the sensor side (MSRV and MSRVIP) since some of the networks have different IP schemes.

> I don't believe we've had any other reports of this, so this might be something specific to your configuration or environment.

The so-setup on login always launches `~/SecurityOnion/setup/so-setup iso` if the host hasn't completed the setup process, but you are correct the issue I'm running into is probably a product of mail already being in /var/spool/mail/root from previous setup runs.

**Reply by dougburks** [reply]
> Would it make sense to add a README to that folder to say this is an unsupported internal testing feature?

Added:
https://github.com/Security-Onion-Solutions/securityonion/blob/dev/setup/automation/README.txt
